### PR TITLE
Remove nested anchors in headers

### PR DESCRIPTION
**Pull request by duch0416**

### CODE REVIEW

**Review by slavik-chapelskyi — COMMENTED** [review]
We are using this on client code, so adding node-parser will drastically bloat our bundlesize
CleanShot 2023-09-29 at 13 44 13@2x

File: src/helpers/getDataWithoutNestedAnchors.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import {parse} from 'node-html-parser';
+import type {TemplateExecutor} from 'lodash';
+import type {Header} from '../types';
+
+const extractAnchorTags = (data: Header) => {
+  // Extract any existing anchor tags from the header variable
+  const matches = data.header.match(/<a[^>]*>.*?<\/a>/g);
```
Comment (slavik-chapelskyi) [inline src/helpers/getDataWithoutNestedAnchors.ts:7]: Also not sure on the performance of this calcualtions + regex lookp over complex html

Comment (duch0416) [reply src/helpers/getDataWithoutNestedAnchors.ts:7]: hmm I can try to parse html first and use querySelectorAll to check for anchor tags what do you think about that ?

Comment (slavik-chapelskyi) [reply src/helpers/getDataWithoutNestedAnchors.ts:7]: Need to check on this, as this might also have a rendering performance issue on gem page

Comment (duch0416) [reply src/helpers/getDataWithoutNestedAnchors.ts:7]: this dosen't check on the whole content its only check for small piece of HTML inside header

Comment (vladholubiev) [reply src/helpers/getDataWithoutNestedAnchors.ts:7]: no, please no regex here.

**Pull request closed without merging**